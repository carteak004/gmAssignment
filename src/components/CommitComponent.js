import React from 'react';
import {View, Text, useColorScheme} from 'react-native';
import {Colors} from 'react-native/Libraries/NewAppScreen';

const CommitComponent = ({history}) => {
  const isDarkMode = useColorScheme() === 'dark';

  const backgroundStyle = {
    backgroundColor: isDarkMode ? Colors.darker : Colors.lighter,
  };

  const {author, hash, message} = history;
  return (
    <View style={backgroundStyle}>
      <Text style={{color: 'red'}}>{author}</Text>
      <Text>{hash}</Text>
      <Text>{message}</Text>
    </View>
  );
};

export default CommitComponent;
