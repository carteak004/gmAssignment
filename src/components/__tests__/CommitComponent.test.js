import React from 'react';
import renderer from 'react-test-renderer';
import CommitComponent from '../CommitComponent';

test('renders correctly', () => {
  const history = {
    hash: '1',
    message: 'created test',
    name: 'Kartheek',
  };
  const tree = renderer.create(<CommitComponent history={history} />).toJSON();
  expect(tree).toMatchSnapshot();
});
