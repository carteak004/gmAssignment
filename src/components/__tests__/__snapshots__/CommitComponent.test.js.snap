// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "flex": 1,
        "padding": 10,
      },
      Object {
        "backgroundColor": "#FFF",
      },
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "fontSize": 24,
          "fontWeight": "600",
        },
        Object {
          "color": "#000",
        },
      ]
    }
  >
    Kartheek
  </Text>
  <Text
    style={
      Array [
        Object {
          "fontSize": 18,
          "fontWeight": "400",
          "marginTop": 8,
        },
        Object {
          "color": "#444",
        },
      ]
    }
  >
    1
  </Text>
  <Text
    style={
      Array [
        Object {
          "fontSize": 16,
          "fontWeight": "700",
          "marginTop": 8,
        },
        Object {
          "color": "#444",
        },
      ]
    }
  >
    created test
  </Text>
</View>
`;
